source/pool/contracts/Pool.sol

```diff
@@ -363,4 +368,41 @@ contract Pool is IPool, Ownable {
     bytes32 leaf = keccak256(abi.encodePacked(participant, score));
     return MerkleProof.verify(proof, root, leaf);
   }
-}
+
+/** @notice withdraw function that uses signature instead of claim calculated from merkle root
+  * @param v signature parameter v of one of the admins
+  * @param r signature parameter r of one of the admins
+  * @param s signature parameter s of one of the admins
+  * @param messageHash hash of token address, minimumAmount, recipient and nonce
+  * @param token address
+  * @param amount uint256
+  * @param nonce add nonce to the hash, nonce must be non-reusable
+  */
+  function withdrawWithSignature(
+    uint8 v,
+    bytes32 r,
+    bytes32 s,
+    bytes32 messageHash,
+    address token,
+    uint256 amount,
+    uint256 nonce
+  ) external override returns (uint256) {
```

Comment:
Since this is an external, state-changing function, it should go above internal and/or pure/view functions as a styling best practice. I'd suggest moving it to just above the withdrawProtected function.